### PR TITLE
ci: verify the solution is correct

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

### CODE REVIEW

File: tests/test_solutions.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+CHALLENGES_DIR = Path(__file__).parent.parent / "challenges"
+ALL_SOLUTIONS = list(CHALLENGES_DIR.glob("**/solution.py"))
+
+
+@pytest.mark.parametrize("solution_file", ALL_SOLUTIONS, ids=lambda x: x.parent.name)
+def test_solution_valid(solution_file: Path):
+    with solution_file.open() as f:
+        code = f.read()
+    result = ChallengeManager._type_check_with_pyright(code)
+    assert result.passed, result.stdout
```
Comment (100gle) [inline tests/test_solutions.py:16]: I think we should test these cases in parallel to reduce the total time of CI. How about [pytest-xdist](https://github.com/pytest-dev/pytest-xdist)? 👀

Comment (frostming) [reply tests/test_solutions.py:16]: here you go

**Review by 100gle — APPROVED** [review]
LGTM!

File: tests/test_solutions.py (hunk 1 above)
Comment (laike9m) [inline tests/test_solutions.py:16]: Curious, is there a need to `assert result.stdout`?

**Comment by laike9m** [comment]
Thanks, I always wanted to add this but don't have time. Overall LGTM. There were some new solutions added just now, could you do a pull and test them? Appreciate it!

File: tests/test_solutions.py (hunk 1 above)
Comment (frostming) [reply tests/test_solutions.py:16]: This will give the typecheck output when the test fails, which helps up debug the solution.

**Comment by frostming** [comment]
> Some new solutions were just added, do you mind do a pull and test them? Appreciate it!

You can rebase the main onto the pull requests after this is merged. New solutions will be automatically picked up for CI, and the author will know the status.

File: tests/test_solutions.py (hunk 1 above)
Comment (laike9m) [reply tests/test_solutions.py:16]: I see, makes sense.

**Comment by laike9m** [comment]
> > Some new solutions were just added, do you mind do a pull and test them? Appreciate it!
> 
> You can rebase the main onto the pull requests after this is merged. New solutions will be automatically picked up for CI, and the author will know the status.

I mean they've been added to the codebase
https://github.com/laike9m/Python-Type-Challenges/commit/73fa81e3c302a021f2f3e616e78b05ed9234b631

**Comment by frostming** [comment]
> I mean they've been added to the codebase
> [73fa81e](https://github.com/laike9m/Python-Type-Challenges/commit/73fa81e3c302a021f2f3e616e78b05ed9234b631)

Oh, I've rebased the main and it is already included in the CI result :)

**Pull request merged**